core/arch/arm/mm/core_mmu_lpae.c

```diff
@@ -642,46 +642,64 @@ void core_mmu_create_user_map(struct user_ta_ctx *utc,
 bool core_mmu_find_table(vaddr_t va, unsigned max_level,
 		struct core_mmu_table_info *tbl_info)
 {
-	uint64_t *tbl = l1_xlation_table[0][get_core_pos()];
-	uintptr_t ntbl;
-	unsigned level = 1;
+	uint32_t exceptions = thread_mask_exceptions(THREAD_EXCP_ALL);
+	unsigned int num_entries = NUM_L1_ENTRIES;
+	unsigned int level = 1;
 	vaddr_t va_base = 0;
-	unsigned num_entries = NUM_L1_ENTRIES;
+	bool ret = false;
+	uintptr_t ntbl;
+	uint64_t *tbl;
 
+	tbl = l1_xlation_table[0][get_core_pos()];
 	while (true) {
 		unsigned level_size_shift =
 			L1_XLAT_ADDRESS_SHIFT - (level - 1) *
 						XLAT_TABLE_ENTRIES_SHIFT;
 		unsigned n = (va - va_base) >> level_size_shift;
 
 		if (n >= num_entries)
-			return false;
+			goto out;
 
 		if (level == max_level || level == 3 ||
 			(tbl[n] & TABLE_DESC) != TABLE_DESC) {
 			/*
 			 * We've either reached max_level, level 3, a block
 			 * mapping entry or an "invalid" mapping entry.
 			 */
+
+			/*
+			 * Level 1 is the CPU specific translation table.
+			 * It doesn't make sense to return anything based
+			 * on that unless foreign interrupts already are
+			 * masked.
+			 */
+			if (level == 1 &&
+			    !(exceptions & THREAD_EXCP_FOREIGN_INTR))
```

Comment:
Is `!(exceptions & THREAD_EXCP_FOREIGN_INTR)` really necessary? Everything is already masked when `exception` is initialized above.

Reply:
`exceptions` gives the state of the exceptions masked before `core_mmu_find_table()` called `thread_mask_exceptions()`. This is what we wanted to test.

Reply:
Ah, ok. Thanks!